tests/actions.go

```diff
@@ -3619,6 +3616,35 @@ func (oa *OperatorActions) WaitForTidbClusterReady(tc *v1alpha1.TidbCluster, tim
 	})
 }
 
+func (oa *OperatorActions) WaitForDmClusterReady(dc *v1alpha1.DMCluster, timeout, pollInterval time.Duration) error {
+	if dc == nil {
+		return fmt.Errorf("DmCluster is nil, cannot call WaitForDmClusterReady")
+	}
+	return wait.PollImmediate(pollInterval, timeout, func() (bool, error) {
+		var (
+			local *v1alpha1.DMCluster
+			err   error
+		)
+		if local, err = oa.cli.PingcapV1alpha1().DMClusters(dc.Namespace).Get(dc.Name, metav1.GetOptions{}); err != nil {
+			log.Logf("failed to get DmCluster: %s/%s, %v", dc.Namespace, dc.Name, err)
+			return false, nil
+		}
+
+		if b := oa.dmMasterMembersReadyFn(local); !b {
+			log.Logf("dm master members are not ready for dc %q", dc.Name)
+			return false, nil
+		}
+		log.Logf("dm master members are ready for dc %q", dc.Name)
+		if b := oa.dmWorkerMembersReadyFn(local); !b {
+			log.Logf("dm worker memebers are not ready for dc %q", dc.Name)
```

Comment:
Do we need to return here?

Reply:
👍 fixed in https://github.com/pingcap/tidb-operator/pull/3946/commits/696b8486a1e349bf8d9dbd21b32c07a33cdbea51